ubi_manifest/worker/tasks/depsolver/modulemd_depsolver.py

```diff
@@ -122,6 +134,33 @@ def _already_searched(self, module):
             or f"{module.name}:{module.stream}" in self._searched_modules["with_stream"]
         )
 
+    def _update_rpm_dependencies(self, module):
+        """
+        Adds all pkgs from artifacts to rpm and debuginfo dependencies. Omits source
+        pkgs and filters by profiles if available.
+        """
+        if module.artifacts:
+            pkg_names = []
+            key = f"{module.name}:{module.stream}"
+            if module.profiles:
+                for profile in self._profiles.get(key) or []:
+                    pkg_names.extend(module.profiles.get(profile) or [])
+
+            for pkg in module.artifacts_filenames:
+                # Filter out source packages
+                if pkg.endswith(".src.rpm"):
+                    continue
+                # filter by profile if available
+                if pkg_names:
+                    name, _, _, _, _ = split_filename(pkg)
+                    if name not in pkg_names:
+                        continue
+
+                if "debuginfo" in pkg or "debugsource" in pkg:
```

Comment:
This seems not safe because you don't know if these words exist in the pkg name because it's a debuginfo/debugsource RPM, or just because the package happens to be named that way.

I checked on my local Fedora system and found there are several packages which genuinely have debuginfo in the name despite not being debuginfo packages, like these:

```
====================================== Name & Summary Matched: debuginfo ======================================
elfutils-debuginfod-client-devel.i686 : Libraries and headers to build debuginfod client applications
elfutils-debuginfod-client-devel.x86_64 : Libraries and headers to build debuginfod client applications
=========================================== Name Matched: debuginfo ===========================================
elfutils-debuginfod.x86_64 : HTTP ELF/DWARF file server addressed by build-id
elfutils-debuginfod-client.x86_64 : Library and command line client for build-id HTTP ELF/DWARF server
elfutils-debuginfod-client.i686 : Library and command line client for build-id HTTP ELF/DWARF server
python3.8-debuginfo.x86_64 : Debug information for package python3.8
```

So for example, looking at https://koji.fedoraproject.org/koji/buildinfo?buildID=1963497 , I believe you would want the RPM named elfutils-debuginfod-debuginfo-0.187-4.fc37.aarch64.rpm to go into this path but would NOT want the RPM named elfutils-debuginfod-0.187-4.fc37.aarch64.rpm to go here.  The current code would wrongly treat both of the RPMs as debuginfo.

The example comes from Fedora.  I don't know whether this code actually needs to process any RPMs which run into this problem, but even if it doesn't today, it seems likely that it would eventually.

Is there a more robust way to divide these into debuginfo and non-debuginfo?

Reply:
And just to add my own idea how this is probably feasible with the data we have today - referencing example https://kojipkgs.fedoraproject.org//packages/ghc/8.4/820200708061905.9edba152/files/module/modulemd.x86_64.txt

For debugsource, it should always be named like ${srcpackagename}-debugsource.  So in the above example, because the SRPM name is "ghc", we can safely conclude that the package named "ghc-debugsource" is the debugsource package.

For debuginfo, it should always be named like ${subpackagename}-debuginfo.  So in the above example, when we see:

1. ghc-base-0:4.11.1.0-74.module_el8+12161+cf1bd7f2.x86_64
2. ghc-base-debuginfo-0:4.11.1.0-74.module_el8+12161+cf1bd7f2.x86_64

We can safely conclude that (2) is the debuginfo package for (1).  On the other hand, if ghc-base-debuginfo existed but ghc-base didn't exist, we could not safely conclude that ghc-base-debuginfo was a debuginfo package, it might instead just be a package whose name happens to end in debuginfo.

Reply:
There is also another way which is used in ubipop for along time that doesn't split the debug packages from artifacts. It's not optimal but it's safe and we don't have to guess:
1) Take all rpm filenames from artifacts
2) Search all of them in binary repo -> only binary returned
3) Search all of them in debug repo -> only debug returned

Reply:
Thanks for the comments! I have implemented the solution Rob has mentioned. 
I could also change it, so that we search for the subpackagename for each debuginfo package to verify, that it is indeed a debuginfo package. But that might have a bigger overhead than not splitting the packages, since for each debuginfo pkg we would have to make a separate pulp query.